src/client/app/containers/import.ts

```diff
@@ -213,4 +219,36 @@ export class Import {
     cancel() {
         this._store.dispatch(new UI.ToggleImportAction(false));
     }
+
+    checkForViewSnippet() {
+        if (!this.hasViewSettings()) {
+            return;
+        }
+
+        let viewData = Office.context.document.settings.get('VIEW_URL_SETTINGS');
+        if (viewData.type === 'samples') {
+            let hostJsonFile = `${environment.current.config.samplesUrl}/view/${environment.current.host.toLowerCase()}.json`;
+            let sub = this._request.get<JSON>(hostJsonFile, ResponseTypes.JSON)
+                .subscribe(lookupTable => {
+                    if (lookupTable && lookupTable[viewData.id]) {
```

Comment:
What is the cache expiration header on the .json file? Will the browser request the file within a few hours of adding a new sample? Should we cache bust this json file?